<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../icons/program.png</normaloff>../icons/program.png</iconset>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="CentralWidget">
   <property name="minimumSize">
    <size>
     <width>320</width>
     <height>150</height>
    </size>
   </property>
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <layout class="QGridLayout" name="gridLayout" columnstretch="1,3">
    <property name="leftMargin">
     <number>5</number>
    </property>
    <property name="topMargin">
     <number>5</number>
    </property>
    <property name="rightMargin">
     <number>5</number>
    </property>
    <property name="bottomMargin">
     <number>5</number>
    </property>
    <item row="3" column="1">
     <widget class="QWidget" name="MessageField" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="5,1">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLineEdit" name="MessageTextEdit">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>50</height>
          </size>
         </property>
         <property name="locale">
          <locale language="English" country="UnitedStates"/>
         </property>
         <property name="maxLength">
          <number>256</number>
         </property>
         <property name="placeholderText">
          <string>Write your message...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="SendButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QWidget" name="InformationWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="CurrentSessionInfo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="locale">
          <locale language="English" country="UnitedStates"/>
         </property>
         <property name="text">
          <string>Current session: {CurrentSessionInfo}</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="SettingsButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="locale">
          <locale language="English" country="UnitedStates"/>
         </property>
         <property name="text">
          <string>Settings</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QListWidget" name="ChatList">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>List of available connections</string>
      </property>
      <property name="whatsThis">
       <string>List of available connections</string>
      </property>
      <property name="showDropIndicator" stdset="0">
       <bool>false</bool>
      </property>
      <property name="horizontalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="AddConnectionButton">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>30</height>
       </size>
      </property>
      <property name="cursor">
       <cursorShape>PointingHandCursor</cursorShape>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string>Add connection</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QPushButton" name="DisconnectButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Disconnect</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QListWidget" name="MessageArea">
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::NoSelection</enum>
      </property>
      <property name="verticalScrollMode">
       <enum>QAbstractItemView::ScrollPerPixel</enum>
      </property>
      <property name="layoutMode">
       <enum>QListView::Batched</enum>
      </property>
      <property name="batchSize">
       <number>2</number>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>MessageTextEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>EnableSendButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>607</x>
     <y>584</y>
    </hint>
    <hint type="destinationlabel">
     <x>848</x>
     <y>496</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>AddConnectionButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowConnectionDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>16</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>460</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SettingsButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowSettingsDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>805</x>
     <y>8</y>
    </hint>
    <hint type="destinationlabel">
     <x>850</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ChatList</sender>
   <signal>itemClicked(QListWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowChatFor(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SendButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>SendMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>766</x>
     <y>576</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>578</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MessageTextEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>SendMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>477</x>
     <y>580</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DisconnectButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>RemoveConnection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>EnableSendButton()</slot>
  <slot>ShowConnectionDialog()</slot>
  <slot>ShowSettingsDialog()</slot>
  <slot>ShowChatFor(QListWidgetItem*)</slot>
  <slot>SendMessage()</slot>
  <slot>RemoveConnection()</slot>
 </slots>
</ui>
